sys/net/gnrc/netif/init_devs/auto_init_at86rf215.c

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2019 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/*
+ * @ingroup sys_auto_init_gnrc_netif
+ * @{
+ *
+ * @file
+ * @brief   Auto initialization for at86rf215 network interfaces
+ *
+ * @author  Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ */
+
+#ifdef MODULE_AT86RF215
+
+/* AT86RF215M only has a sub-GHz interface */
+#define USE_DUALBAND !(IS_USED(AT86RF215_DISABLE_SUBGHZ) || IS_USED(MODULE_AT86RF215M))
```

Comment:
This has an issue now, it leads to an assert in `sys/net/gnrc/netif/gnrc_netif.c:64 => 0x200e8d` since `GNRC_NETIF_NUMOF` is 1 by default.

I think this configuration still leaves some weird cases:

- GNRC_NETIF_NUMOF = 1
  - only subghz: compile as is
  - only 2.4ghz: `CFLAGS=-DAT86RF215_DISABLE_SUBGHZ`
  - both: should not be possible

- GNRC_NETIF_NUMOF > 1 (we assume here that there are enough for all interfaces the user might want)
  - only subghz: should be possible, is not
  - only 2.4ghz: `CFLAGS=-DAT86RF215_DISABLE_SUBGHZ`
  - both: compile as is

IMO this could be handled better with `PSEUDOMODULES` `AT86RF215_SUBGHZ` `AT86RF215_24GHZ`. This way if `GNRC_NETIF_NUMOF=1` and both are included you could print a compilation error instead of having the assert jump.

What do you think?

Reply:
I like the idea with the submodules! Makes everything much cleaner.
The only thing I don't like is `AT86RF215_24GHZ` because that's an order of magnitude over the frequency range supported by the radio, but I can live with that 😉